src/Workspaces/Core/Portable/Options/OptionSet.cs

```diff
@@ -93,11 +93,11 @@ bool IOptionsReader.TryGetOption<T>(OptionKey2 optionKey, out T value)
     /// Checks if the value is an internal representation -- does not cover all cases, just code style options.
     /// </summary>
     internal static bool IsInternalOptionValue(object? value)
-        => value is not ICodeStyleOption codeStyle || ReferenceEquals(codeStyle, codeStyle.AsInternalCodeStyleOption());
+        => value is not ICodeStyleOption codeStyle || codeStyle.GetType() == typeof(CodeStyleOption2<>);
```

Comment:
This is never going to match (need to get the generic type definition on the left side)